File: other/Solution26.java
class Solution26 {
	public int removeDuplicates(int[] nums) {
		if(nums.length <= 1){
			return nums.length;
		}
		int i = 0;
		int j = 1;
		int length = 1;
		while(j < nums.length){
			if(nums[i] == nums[j]){
				j++;
			}
			else{
				length++;
				i++;
				nums[i] = nums[j];
			}

		}return length;
	}
}

File: C++/Solution26.cpp
#include<iostream>
#include<vector>

using namespace std;

class Solution26 {
public:
    int removeDuplicates(vector<int>& nums) {
        int size = nums.size();
        int index1 = 0, index2 = 0;
        while(index2 < size){
            while(index2 < size && nums[index2] == nums[index1]){
                index2++;
            }
            if(index2 < size){
                nums[index1 + 1] = nums[index2];
            }
            else{
                return index1 + 1;
            }
            index1++;
        }
        return index1 + 1;
    }
};